debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -117,38 +136,97 @@ private PostgresReplicationConnection(Configuration config,
+                                            Set<TableId> tablesToCapture = determineCapturedTables(new RelationalSnapshotChangeEventSource.RelationalSnapshotContext(""));
+                                            String tableFilterString = tablesToCapture.stream().map(TableId::toDoubleQuotedString).collect(Collectors.joining(", "));
+                                            // Publication doesn't exist, create it but restrict to the tableFilter.
+                                            stmt.execute(String.format("CREATE PUBLICATION %s FOR TABLE %s;", publicationName, tableFilterString));
+                                        }
+                                        catch (Exception e) {
+                                            LOGGER.error("Unable to create publication for ");
+                                        }
+                                        break;
+                                }
+                            }
+                            else {
+                                LOGGER.trace(
+                                        "A logical publication named '{}' for plugin '{}' and database '{}' is already active on the server " +
+                                                "and will be used by the plugin",
+                                        publicationName, plugin, database());
+                            }
                         }
                     }
                 }
+                catch (SQLException e) {
+                    throw new JdbcConnectionException(e);
+                }
+
+                // This is what ties the publication definition to the replication stream
+                streamParams.put("proto_version", 1);
+                streamParams.put("publication_names", publicationName);
             }
-            catch (SQLException e) {
-                throw new JdbcConnectionException(e);
-            }
+        }
+    }
+
+    private Stream<TableId> toTableIds(Set<TableId> tableIds, Pattern pattern) {
+        return tableIds
+                .stream()
+                .filter(tid -> pattern.asPredicate().test(tid.toString()))
+                .sorted();
+    }
+
+    private Set<TableId> sort(Set<TableId> capturedTables) throws Exception {
+        String value = config().getString(RelationalDatabaseConnectorConfig.TABLE_WHITELIST);
+        if (value != null) {
+            return Strings.listOfRegex(value, Pattern.CASE_INSENSITIVE)
+                    .stream()
+                    .flatMap(pattern -> toTableIds(capturedTables, pattern))
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+        }
+        return capturedTables
+                .stream()
+                .sorted()
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    private Set<TableId> determineCapturedTables(RelationalSnapshotChangeEventSource.RelationalSnapshotContext ctx) throws Exception {
```

Comment:
Why is the context parameter needed here?

Reply:
that's the part where I wasn;t really sure the best way of using the table filter contract. since based on `RelationalSnapshotChangeEventSource.determineCapturedTables(RelationalSnapshotContext)` it appears that is needs to know all the tables and then iterate through and build up a list of TableIds that match

Reply:
actually I may have just figure out how that works...should have  some updates later today! thanks so much for helping me through on this!